src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs

```diff
@@ -250,6 +253,9 @@ public int LaunchCustomHost(TestProcessStartInfo testProcessStartInfo)
                     waitHandle.Set();
                 };
 
+                // Registering cancellationToken to set waitHandle (whenever request is cancelled).
+                var cancellationTokenRegistration = cancellationToken.Register(() => waitHandle.Set());
```

Comment:
nit: would something like [this](https://stackoverflow.com/questions/12129749/cancellation-with-waithandle) be more direct? Msdn usage [here](https://docs.microsoft.com/en-us/dotnet/standard/threading/how-to-listen-for-cancellation-requests-that-have-wait-handles)

Reply:
done.